### ISSUE TITLE
How to test docs locally

### ISSUE THREAD

**Issue by patrickvonplaten** [huggingface/doc-builder#57]
Can we leave a short README.md here how to test the docs locally?

**Comment by patrickvonplaten** [comment]
Describing here a bit approach on how to run the docs locally.

I recommend doing the following:

1. Git clone moon-landing
2. Install all necessary requirements for moon-landing
3. Start all services for moon-landing
4. Get the docs running with moon-landing

We assume you are in the home folder `/home/user1/`
For 1.:
```
git clone https://github.com/huggingface/moon-landing.git
```

For 2.:
- Install all requirements as stated here: https://github.com/huggingface/moon-landing#requirements
- Also follow the steps here: https://github.com/huggingface/moon-landing#git-submodules

For 3.:
The services that you need to start are: 
- mongodb: Run ```sudo systemctl start mongod```
- redis: Run ```redis-server``

For 4.:
- First, clone the doc-builder in the `repos` repository of moon-landing:
```bash
cd /home/user1/moon-landing/repos && git clone https://github.com/huggingface/doc-builder.git
```
- Now install all required dependencies of the doc-builder
```bash
cd /home/user1/moon-landing/repos/doc-builder && pip install -e .
```
- Now you can build the docs from local docs of a transformers repo. Let's assume the transformers repo: https://github.com/huggingface/transformers/tree/master/src/transformers is under `/home/user1/transformers`.
Then you should run the following command:
```
cd /home/user1/moon-landing/repos/ && doc-builder build transformers /home/user1/transformers/docs/source --build_dir doc-builder/build --notebook_dir notebooks/transformers_doc --clean
```
**This is the most important step**: It will create the docs depending on your local configuration of transformers. So whenever you add new docs and want to test them locally. You should just make sure that the path `/home/user1/transformers/docs/source` points to your locally adapted transformers repo. Also make sure that the official python path of `transformers` points to this repo.

You should see in your terminal how the docs are built
- Now you can start the server. Following the steps here: https://github.com/huggingface/moon-landing#compile-in-watch-mode
In short you should do the following:
1.
```
cd /home/user1/moon-landing/ && git pull --recurse-submodules
```
2.
```
cd /home/user1/moon-landing/front && npm i
```
3.
```
cd /home/user1/moon-landing/server && npm i

```
4.
```
cd /home/user1/moon-landing/server && npm run watch
```
Now you should see how the docs are actually built. This will take some time (a couple of minutes).
Once this step is finished you can check out your docs under:
```
http://localhost:5564/docs/transformers/master/en
```

**Comment by patrickvonplaten** [comment]
cc @sgugger @mishig25 - thanks for your pointers. Maybe this can help as a "I-only-know-transformer" guide to how to test the docs locally.

**Comment by LysandreJik** [comment]
Thanks for the extremely useful guide! I think there's a command missing in your guide, specifying to install `transformers[dev]`

The following step was also missing for me: `node build-svelte.mjs --doc`, it showed a 404 while this wasn't run.

**Comment by patrickvonplaten** [comment]
Thanks! Do you want to update the notion here: https://www.notion.so/huggingface2/Running-Transformers-Docs-Locally-e64aaaed1e884abf8225fbc2ac0af3a4 ? 

I kind of assumed that `transformers` was already installed with dev in the Python environment since this is mainly for "transformers"-people, but it's definitely better to mention how to install it in case it's not.

```node build-svelte.mjs --doc``` -> sorry yeah I think I forgot that command

**Comment by patrickvonplaten** [comment]
**UPDATE** (better approach):

Describing here my approach on how to run the docs locally. This might be especially useful for people coming from the open-source `transformers` team with limited front-end XP (like me).

I recommend doing the following:

1. Git clone moon-landing
2. Install all necessary requirements for moon-landing
3. Start all services for moon-landing
4. Get the docs running with moon-landing

We assume you are in the home folder `/home/user1/`
**1.)**
```bash
git clone https://github.com/huggingface/moon-landing.git
```

**2.)**
- Install all requirements as stated here: https://github.com/huggingface/moon-landing#requirements
- Also follow the steps here: https://github.com/huggingface/moon-landing#git-submodules

**3.)**
The services that you need to start are: 
- mongodb: Run ```sudo systemctl start mongod```
- redis: Run ``redis-server``

**4.)**

Now I would link you local clone of `transformers` and `datasets` to `moon-landing/repos`, *e.g.*:

```bash
cd ~/moon-landing && ln -s $(realpath ~/transformers) ./
cd ~/moon-landing && ln -s $(realpath ~/datasets) ./
```

Now you can follow the steps here: https://github.com/huggingface/moon-landing#building-the-libraries-doc-eg-docstransformers-and-more-libraries-to-come 

I strongly recommend  directly adapting the files in this folder to see the direct changes live in your browser - *e.g.* if you are not sure about how your written docs look like.
The very nice thing is that because you are in watch mode this way the docs don't need to be rebuilt.
